src/modules/aws_rds/mappers/db_cluster.ts

```diff
@@ -330,6 +330,12 @@ export class DBClusterMapper extends MapperBase<DBCluster> {
     delete: async (es: DBCluster[], ctx: Context) => {
       for (const e of es) {
         const client = (await ctx.getAwsClient(e.region)) as AWS;
+
+        // check if deletion protection is enabled and throw error
+        if (e.deletionProtection)
+          throw new Error(
+            "Cannot delete a cluster with deletion protection. Please set the cluster's deletion protection to false and try again.",
+          );
```

Comment:
```suggestion
        if (e.deletionProtection) {
          throw new Error(
            "Cannot delete a cluster with deletion protection. Please set the cluster's deletion protection to false and try again.",
          );
        }
```

Reply:
also, should we `throw` immediately or should we `continue` and `throw` later?

Reply:
you mean, throwing after all other entries have been processed?

Reply:
yes, I think we do not do it in other mappers, but thinking about it, it might be the right way to do it. That way we process what we can always. What do you think?